--- a/src/server.c
+++ b/src/server.c
@@ -309,16 +309,17 @@
 #endif
 
 #ifdef HAVE_CRYPTO
+		if ((server->auth_flags & NVNC_AUTH_ALLOW_BROKEN_CRYPTO) &&
+		    !(server->auth_flags & NVNC_AUTH_REQUIRE_USERNAME) &&
+		    !(server->auth_flags & NVNC_AUTH_REQUIRE_ENCRYPTION)) {
+			ADD_SECURITY_TYPE(RFB_SECURITY_TYPE_VNC_AUTH);
+		}
+
 		ADD_SECURITY_TYPE(RFB_SECURITY_TYPE_RSA_AES256);
 		ADD_SECURITY_TYPE(RFB_SECURITY_TYPE_RSA_AES);
 
 		if (!(server->auth_flags & NVNC_AUTH_REQUIRE_ENCRYPTION)) {
 			ADD_SECURITY_TYPE(RFB_SECURITY_TYPE_APPLE_DH);
-
-			if ((server->auth_flags & NVNC_AUTH_ALLOW_BROKEN_CRYPTO) &&
-			    !(server->auth_flags & NVNC_AUTH_REQUIRE_USERNAME)) {
-				ADD_SECURITY_TYPE(RFB_SECURITY_TYPE_VNC_AUTH);
-			}
 		}
 #endif
 	} else {
